src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.h

```diff
@@ -0,0 +1,688 @@
+};
+
+/** @brief
+ *  Defines interfaces for implementing application-specific logic for various aspects of the PushAvStreamTransport Delegate.
+ *  Specifically, it defines interfaces for the command handling and loading of the allocated streams.
+ */
+class PushAvStreamTransportDelegate
+{
+public:
+    PushAvStreamTransportDelegate() = default;
+
+    virtual ~PushAvStreamTransportDelegate() = default;
+
+    void SetEndpointId(EndpointId aEndpoint) { mEndpointId = aEndpoint; }
+
+    /**
+     *   @brief Handle Command Delegate for stream transport allocation with the provided transport configuration option.
+     *
+     *   @param transportOptions[in]        represent the configuration options of the transport to be allocated
+     *
+     *   @param connectionID[in]            Indicates the connectionID to allocate.
+     *
+     *   @return Success if the allocation is successful and a PushTransportConnectionID was produced; otherwise, the command SHALL
+     * be rejected with Failure.
+     *
+     *   The buffers storing URL, Trigger Options, Motion Zones, Container Options is owned by the PushAVStreamTransport Server.
+     *   The buffers are allocated on success of AllocatePushTransport and deallocated on success of DeallocatePushTransport
+     * command. The delegate is expected to process the transport following transport options: URL :  Validate the URL
+     *   StreamUsage,VideoStreamID,AudioStreamID for selection of Stream.
+     *   Allocate the transport and map it to the connectionID.
+     *   On Success TransportConfigurationStruct is sent as response by the server.
+     */
+    virtual Protocols::InteractionModel::Status AllocatePushTransport(const TransportOptionsStruct & transportOptions,
+                                                                      const uint16_t connectionID) = 0;
+    /**
+     *   @brief Handle Command Delegate for Stream transport deallocation for the
+     *   provided connectionID.
+     *
+     *   @param connectionID[in]        Indicates the connectionID to deallocate.
+     *
+     *   @return Success if the transport deallocation is successful; otherwise, the delegate is expected to return status code BUSY
+     *   if the transport is currently uploading.
+     *
+     */
+    virtual Protocols::InteractionModel::Status DeallocatePushTransport(const uint16_t connectionID) = 0;
+    /**
+     *   @brief Handle Command Delegate for Stream transport modification.
+     *
+     *   @param connectionID [in]           Indicates the connectionID of the stream transport to modify.
+     *
+     *   @param transportOptions [out]      represents the Transport Options to modify.
+     *
+     *   The buffers storing URL, Trigger Options, Motion Zones, Container Options is owned by the PushAVStreamTransport Server.
+     *   The allocated buffers are cleared and reassigned on success of ModifyPushTransport and deallocated on success of
+     * DeallocatePushTransport.
```

Comment:
This doesn't make sense.  This is being handed `TransportOptionsStruct::DecodableType`, no?  That thing points its buffers into the received message, no?

I really don't understand what these comments are trying to say.